File: apps/nex-force/src/types/project.ts
// ============ Project ============
export interface ProjectResponse {
    projectId: string;
    projectName: string;
    project: string;
    deadline: string;
    priority: string;
    description: string;
    startDate: string;
    id: number;
    isActive: boolean;
    projectCode: string;
    approver: string;
    auditor: string;
    inchargeName: string;
    ivNo: string;
    ivDate: string;
    teamLead: TeamMemberResponse;
    team: TeamMemberResponse[];
    files?: ProjectFileDTO[];
}

export interface ProjectResponseDTO {
    projectId: number;
    projectName?: string;
    client?: number;
    startDate?: string | null;
    endDate?: string | null;
    rate?: string;
    rateType?: string;
    priority?: string;
    projectLeader?: number;
    description?: string;
    isActive?: boolean;
    team?: string;
    createDate?: string | null;
    createBy?: string;
    updateDate?: string | null;
    updateBy?: string;
    projectTypeId?: number;
    projectCode?: string;
    inchargeName?: string;
    auditor?: string;
    approver?: string;
    ivDate?: string | null;
    ivNo?: string;
    poNo?: string;
    timesheetDateStart?: number;
    files: ProjectFileDTO[];
}

export interface ProjectView {
    projectId: number;
    startDate?: string | null;
    deadline?: string | null;
    priority?: string;
    isActive?: boolean;
    description?: string;
    id: number;
    projectCode?: string;
    inchargeName?: string;
    auditor?: string;
    approver?: string;
    ivDate?: string | null;
    timesheetDateStart?: number;
    ivNo?: string;
    poNo?: string;
    projectTypeId?: number;
    teamLead?: TeamMemberResponse | null;
    team?: TeamMemberResponse[] | null;
    projectName?: string;
    updateDate: string;
    files: ProjectFileDTO[];
    imageFiles: ProjectFileDTO[];
}

export interface TeamMemberResponse {
    employeeId: string;
    id: number;
    firstNameEn: string;
    lastNameEn: string;
    imgPath: string;
}

export interface ProjectType {
    projectTypeId: number;
    id: number;
    projectTypeNameTh: string;
    projectTypeNameEn: string;
    projectTypeCode: string;
    isActive: boolean | null;
    createDate: string | null;
    createBy: string | null;
    updateDate: string | null;
    updateBy: string | null;
}

export interface ProjectFileDTO {
    fileId: number;
    fileCategory?: string;
    originalName: string;
    storedName?: string;
    filePath?: string;
}

export interface AssignUserRequest {
    projectId: number;
    employeeId: number;
    roleType?: string;
}

// ============ Jobs ============
export interface ManageJobs {
    manageJobId: number;
    jobTitle: string;
    department: string;
    location: string;
    noOfVacancies: number;
    experience: string;
    salaryFrom: number;
    salaryTo: number;
    jobType: string;
    status: string;
    startDate: string;
    expiredDate: string;
    description: string;
    createDate?: string;
    createBy?: string;
    updateDate?: string;
    updateBy?: string;
    id?: number;
}

export interface ManageJobsData {
    data: ManageJobs[];
    totalData: number;
}

export interface JobTypeDto {
    employee_type_id: number;
    employee_type_code: string | null;
    employee_type_name_en: string | null;
}

export interface LocationJobDto {
    jobLocationId: number;
    client_id: number;
    client_code: string | null;
    client_name_en: string | null;
}

export interface PositionJobDto {
    designation_id: number;
    designation_code: string | null;
    designation_name_en: string | null;
}

File: services/nex-force-api/Middleware/Models/Project.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Middlewares.Models
{
    [Table("emp-tb-ms-projects")]
    public class Project
    {
        public int ProjectId { get; set; }
        public string? ProjectName { get; set; }
        public int? Client { get; set; }
        public DateTime? StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        public string? Rate { get; set; }
        public string? RateType { get; set; }
        public string? Priority { get; set; }
        public int? ProjectLeader { get; set; }
        public string? Description { get; set; }
        public bool? IsActive { get; set; }
        public string? Team { get; set; }
        public DateTime? CreateDate { get; set; }             // Creation date
        public string? CreateBy { get; set; }                 // Creator's name
        public DateTime? UpdateDate { get; set; }             // Update date
        public string? UpdateBy { get; set; }
        public decimal? ProjectTypeId { get; set; }
        public string? ProjectCode { get; set; }
        public string? InchargeName { get; set; }
        public string? Auditor { get; set; }
        public string? Approver { get; set; }
        public DateTime? IvDate { get; set; }
        public string? IvNo { get; set; }
        public string? PoNo { get; set; }
        public int? TimesheetDateStart { get; set; }
    }
}
